packages/stream/solana/StreamClient.ts

```diff
@@ -450,16 +563,29 @@ export class SolanaStreamClient extends BaseStreamClient {
       metadata: Keypair | undefined;
       recipient: string;
     }[] = [];
-    metadataPubKeys = metadataPubKeys || [];
+    const metadataPubKeys = metadataPubKeysExt || [];
 
     for (let i = 0; i < recipients.length; i++) {
       const recipientData = recipients[i];
-      const { ixs, metadata, metadataPubKey } = await this.prepareStreamInstructions(recipientData, data, {
-        sender,
-        metadataPubKeys: metadataPubKeys[i] ? [metadataPubKeys[i]] : undefined,
-        computePrice,
-        computeLimit,
-      });
+
+      // here we change logic in case we are doing an alignedUnlock!!!
+      const { ixs, metadata, metadataPubKey } = !!alignedConfigParams
```

Comment:
I wouldn't split it because of `get`, maybe keep 1 client for stream, but make separate "clients" (or how to call them) for instruction "preparation" which can be passed to the function

Reply:
We don't add different functionality here for aligned vs not, only change 1 function call, don't see need for Mixins.

I am strongly for having a single client for simplicity, but open to other proposals

Reply:
Are these streams (aligned) the same entity type on the chain? 

if yes, I'd say that this `createMultiple` could handle both, but in the more explicit way - 
for instance, 
`data` argument could accept 2 types of params - RegularStream and AlignedStream. Based on the input type this function could decide. 

Btw, does it make sense to allow `createMultiple` to create different types with a single call? ex. 2 aligned and 1 regular

Reply:
UPD: 
I see we have N additional APIs (create/cancel) which have these switches. I think it makes sense to have 

AlignedContractsAdapter (or whatever name fits) which accepts SolanaStreamClient instance (and btw in theory it can implement GenericStreamClient interface too)

like:
`new AlignedContractsAdapter(solanaStreamClient)`,
with this delegation all capabilities from the original client class are available, and you can implement all necessary addons on top of it

Reply:
1st comment - the Stream Contracts themselves are the same entity, but instead of directly creating the Contract we create the proxy which then creates the contract. The streams take the exact same params + the aligned configs params (which are passed as creation params for all streams). tbh I don't see the point of adding ability to create both aligned and non-aligned in same call, we don't use it ourselves, unlikely someone else will.

Reply:
2nd comment - I wouldn't say 2 is N :D 

I don't really follow the Adapter story, it would have to change way `create` methods inside of `client` class work and that doesn't make sense.